caikit_nlp/modules/token_classification/filtered_span_classification.py

```diff
@@ -321,19 +321,45 @@ def _stream_span_output(self, text_stream):
         stream_accumulator = ""
         detected_spans = None
         detected_span_count = 0
+        span_start_offset = 0
+
+        def __update_spans(token):
+            # Fix offset since we are restarting stream accumulator every-time.
+            # (See comment below)
+            token.start += span_start_offset
+            token.end += span_start_offset
+            return token
+
         for text in text_stream:
-            # FIXME: stream_accumulator is an ever increasing, so this will
-            # slow down tokenization as the stream becomes filled with more
-            # characters. However, this is done to get around the offset problem
             stream_accumulator += text
             detected_spans = self.tokenizer.run(stream_accumulator).results
 
-            if len(detected_spans) > (detected_span_count + 1):
+            if len(detected_spans) > 1:
+
+                # we have detected more than 1 sentences,
+                # return 1st sentence
+                new_span = detected_spans.pop(0)
+
+                new_span = __update_spans(new_span)
+
                 # we have detected new sentence, return the new sentence
-                yield detected_spans[detected_span_count]
+                yield new_span
 
                 detected_span_count += 1
 
+                # Reset stream accumulator to optimize detection.
+                # this way we don't have to keep tokenizing already tokenized
+                # sentences
+                stream_accumulator = text
+
+                # since we are resetting the accumulator, this means
+                # the spans returned by the tokenizer will all now
+                # start last token offset + 1. So we store the last processed count
```

Comment:
hm "count" seems a bit misleading here - more like index?

also missing word?
```suggestion
                # start with last token offset + 1. So we store the last processed count
```